data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/ListToMapProcessor.java

```diff
@@ -0,0 +1,144 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.opensearch.dataprepper.logging.DataPrepperMarkers.EVENT;
+
+@DataPrepperPlugin(name = "list_to_map", pluginType = Processor.class, pluginConfigurationType = ListToMapProcessorConfig.class)
+public class ListToMapProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger LOG = LoggerFactory.getLogger(ListToMapProcessor.class);
+    private final ListToMapProcessorConfig config;
+
+    @DataPrepperPluginConstructor
+    public ListToMapProcessor(final PluginMetrics pluginMetrics, final ListToMapProcessorConfig config) {
+        super(pluginMetrics);
+        this.config = config;
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        for (final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+
+            final JsonNode sourceNode;
+            try {
+                sourceNode = getSourceNode(recordEvent);
+            } catch (final Exception e) {
+                LOG.error(EVENT, "Given source path [{}] is not valid on record [{}]",
+                        config.getSource(), recordEvent, e);
+                continue;
+            }
+
+            ObjectNode targetNode = null;
+            try {
+                targetNode = constructTargetNode(sourceNode);
+            } catch (IllegalArgumentException e) {
+                LOG.error(EVENT, "Cannot find a list at the given source path [{}} on record [{}]",
+                        config.getSource(), recordEvent, e);
+                continue;
+            } catch (final Exception e) {
+                LOG.error(EVENT, "Error converting source list to map on record [{}]", recordEvent, e);
+            }
+
+            try {
+                updateEvent(recordEvent, targetNode);
+            } catch (final Exception e) {
+                LOG.error(EVENT, "Error updating record [{}] after converting source list to map", recordEvent, e);
+            }
+        }
+        return records;
+    }
+
+    private JsonNode getSourceNode(final Event recordEvent) {
```

Comment:
I suspect the code would be simpler if we returned `Map` instead of `JsonNode`. This is not a necessary change since it is internal to the code.

Reply:
Do you mean a List of Map here? Yeah, looking back it should be easier this way. I probably thought I needed to go deeper along the tree. Thanks for pointing that out!

Reply:
Yes, that is what I meant: `List<Map<String, Object>>`. Again, this is all internal and could be changed later with ease.